### PR TITLE
doc: provide documentation for tsconfigs

### PR DESCRIPTION
**Pull request by wiese**
We separated the tsconfig files. This provides a little context why and
what this individual files are responsible for.

### CODE REVIEW

File: docs/typescript.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This project uses [TypeScript](https://en.wikipedia.org/wiki/TypeScript).
+
+The fact that different parts of this project are built for different target architectures, namely browsers and node servers, is reflected in the TypeScript configuration. The different targets are organized in individual files. Notable differences are the `src/` files included and libraries available.
+
+* `tsconfig.base.json` The base configuration that is inherited by all other configurations
+* `tsconfig.server.json` The configuration used to build for the server
+* `tsconfig.json` The configuration used to build for the client
+    > This file should really be called `tsconfig.client.json` but this is currently not possible as not all tools (which one?) allow for the tsconfig name to be configurable
```
Comment (wiese) [inline docs/typescript.md]: @bitPogo @jakobw Which?

Comment (jakobw) [reply docs/typescript.md]: the vue-cli typescript plugin is the issue, I believe.
https://cli.vuejs.org/config/#typescript
https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-typescript/index.js#L35

Comment (wiese) [reply docs/typescript.md]: Thanks

File: docs/typescript.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+This project uses [TypeScript](https://en.wikipedia.org/wiki/TypeScript).
+
+The fact that different parts of this project are built for different target architectures, namely browsers and node servers, is reflected in the TypeScript configuration. The different targets are organized in individual files. Notable differences are the `src/` files included and libraries available.
+
+* `tsconfig.base.json` The base configuration that is inherited by all other configurations
```
Comment (bitPogo) [inline docs/typescript.md]: Maybe we can repeat here which dir is responsible for that. (e.g. tsconfig.server hits src/server and types/server)...and so on. Just to make things more explicit...but this is totally optional.

Comment (wiese) [reply docs/typescript.md]: Tried that.

**Comment by wiese** [comment]
@jakobw Don't be bothered too much as this PR is not targeting master.

**Pull request merged**